arbor/threading/threading.hpp

```diff
@@ -97,10 +97,27 @@ class task_system {
 
 class task_group {
 private:
+    struct exception_struct {
+        std::atomic<bool> bail{false};
+        std::exception_ptr exception_ptr_;
+        std::mutex exception_mutex_;
+
+        bool have_exception() {
+            return bail.load(std::memory_order_relaxed);
+        }
+
+        void set_exception(std::exception_ptr ex) {
```

Comment:
Rename to `set`? This type has only one job: to manage exception state, so `exception_state::set` is suitably descriptive.